Fix: Setting maxLength to 0 in TextInput still allows typing on iOS

riteshshukla04 opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. The three fields below are mandatory. -->

## Summary:

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
Trying to fix https://github.com/facebook/react-native/issues/52860
## Changelog:

<!-- Help reviewers and the release process by writing your own changelog entry.

Pick one each for the category and type tags:

[ANDROID|GENERAL|IOS|INTERNAL] [BREAKING|ADDED|CHANGED|DEPRECATED|REMOVED|FIXED|SECURITY] - Message

For more details, see:
https://reactnative.dev/contributing/changelogs-in-pull-requests
-->
[IOS][FIXED] Setting maxLength to 0 in TextInput still allows typing on iOS
## Test Plan:

https://github.com/user-attachments/assets/56549e0f-6bbf-461e-815c-794abdee2018

Tested on Android too

riteshshukla04 commented:
I was thinking about this. We could also intialise maxLength with a  negative integer and change the condition `props.maxLength>=0` .

cipolleschi commented:
currently, what is the initial value of `maxLength`? Isn't it better to just set it to Int::Max like explained [here](https://stackoverflow.com/a/8690584)?

riteshshukla04 commented:
Hey @cipolleschi ,
I have updated to Max int. It seems to be working fine . Thanks for the suggestion.
I have also updated `props.maxLength != std::numeric_limits<int>::max()` so that the condition doesn't run everytime.

cipolleschi commented on packages/react-native/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm:
```diff
@@ -450,7 +451,7 @@ - (NSString *)textInputShouldChangeText:(NSString *)text inRange:(NSRange)range
   }
 
-  if (props.maxLength) {
+  if (props.maxLength < std::numeric_limits<int>::max()) {
```
shouldn't this be > 0?
